src/BloomExe/UrlPathString.cs

```diff
@@ -25,8 +26,15 @@ public static UrlPathString CreateFromUnencodedString(string unencoded)
 			// essentially didn't trust that the string was already decoded.
 			// Assuming that was done for a good reason, that behavior is
 			// formalized here. It would seem to be a small risk (makes it
-			// impossible to have, say "%20" in your actual file name)
-			unencoded = HttpUtility.UrlDecode(unencoded);
+			// impossible to have, say "%20" in your actual file name).
+			// However, a '+' in the name is much more likely, and so blindly
+			// re-encoding is a problem. Not clear what's best, so at the moment
+			// I'm going with detecting %20 an only encoding if we see that.
```

Comment:
Sorry, but what you ended up doing is looking for several other possibilities besides %20